scipy/optimize/_differentialevolution.py

```diff
@@ -364,13 +364,16 @@ def __init__(self, func, bounds, args=(),
         self.__scale_arg1 = 0.5 * (self.limits[0] + self.limits[1])
         self.__scale_arg2 = np.fabs(self.limits[0] - self.limits[1])
 
-        parameter_count = np.size(self.limits, 1)
+        self.parameter_count = np.size(self.limits, 1)
+
         self.random_number_generator = _make_random_gen(seed)
 
-        #default initialization is a latin hypercube design, but there
-        #are other population initializations possible.
-        self.population = np.zeros((popsize * parameter_count,
-                                    parameter_count))
+        # default population initialization is a latin hypercube design, but
+        # there are other population initializations possible.
+        self.population = np.zeros((popsize * self.parameter_count,
+                                    self.parameter_count))
```

Comment:
What's the point of initializing `self.population` here? Unless I'm mistaken, It gets overridden unconditionally below by `self.init_population_*` calls.